api/v1/appchart_types.go

```diff
@@ -73,6 +73,7 @@ type AppChartStatus struct {
 	// Important: Run "make" to regenerate code after modifying this file
 }
 
+// Same as ServiceSetting
```

Comment:
The comment is not very explanatory, especially because we are going to have a circular comment dependency:

```
Same as ServiceSetting -> Same as AppChartSetting
```

Since they are the same I would extract this struct to a generic `Setting` object and alias the two types:

```
// api/v1/setting.go

// Setting defines the value and its constraints of a user provided value
type Setting struct {
	// Type of the setting (string, bool, number, or integer)
	Type string `json:"type"`
	// Minimal allowed value, for number, integer
	Minimum string `json:"minimum,omitempty"`
	// Maximal allowed value, for number, integer
	Maximum string `json:"maximum,omitempty"`
	// Enumeration of allowed values, for types string, number, integer
	Enum []string `json:"enum,omitempty"`
	// Presence of an enum for number and integer overrides the min/max
	// specifications
}
```

and then we should be able to just define the two with:

```
type AppChartSetting Setting
type ServiceSetting Setting
```

this will avoid duplication that could be also error prone.

Reply:
I somehow thought the the different CRD types would be in different packages.
Checking I saw it is the same `v1` package.
With that sharing a common structure is not only possible, it also makes much more sense.
Not sure if we should alias them.

Reply:
Handled with commit b0efcf5005d304be9d0e995e4f73118a5217113b